hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java

```diff
@@ -358,6 +359,63 @@ public void testCreateNamespace() throws Exception {
         "There no same location for db and namespace", database2.getLocationUri(), hiveLocalDir);
   }
 
+  @Test
+  public void testCreateNamespaceWithOwnership() throws Exception {
+    Map<String, String> prop =
+        ImmutableMap.of(
+            TableProperties.HMS_DB_OWNER,
+            "apache",
+            TableProperties.HMS_DB_OWNER_TYPE,
+            PrincipalType.USER.name());
+
+    String expectedOwner = "apache";
+    PrincipalType expectedOwnerType = PrincipalType.USER;
+
+    createNamespaceAndVerifyOwnership("userOwnership", prop, expectedOwner, expectedOwnerType);
+
+    prop =
+        ImmutableMap.of(
+            TableProperties.HMS_DB_OWNER,
+            "iceberg",
+            TableProperties.HMS_DB_OWNER_TYPE,
+            PrincipalType.GROUP.name());
+    expectedOwner = "iceberg";
+    expectedOwnerType = PrincipalType.GROUP;
+
+    createNamespaceAndVerifyOwnership("groupOwnership", prop, expectedOwner, expectedOwnerType);
+
+    prop = ImmutableMap.of(TableProperties.HMS_DB_OWNER, "someone");
+    expectedOwner = "someone";
+    expectedOwnerType = PrincipalType.USER; // default is user if not specified
+
+    createNamespaceAndVerifyOwnership("ownedBySomeone", prop, expectedOwner, expectedOwnerType);
+
+    prop = ImmutableMap.of();
+    expectedOwner = System.getProperty("user.name"); // default value if not specified
+    expectedOwnerType = PrincipalType.USER; // default is user if not specified
+
+    createNamespaceAndVerifyOwnership("ownedByDefaultUser", prop, expectedOwner, expectedOwnerType);
+
+    prop = ImmutableMap.of(TableProperties.HMS_DB_OWNER_TYPE, PrincipalType.GROUP.name());
```

Comment:
Just thinking out loud, but shouldn't we indicate failure if only the owner type is set without the owner, instead of silently omitting it?

Reply:
Yes, let's add a precondition check to forbid setting owner type without setting owner.